src/v/security/audit/schemas/tests/ocsf_schemas_test.cc

```diff
@@ -643,3 +653,357 @@ BOOST_AUTO_TEST_CASE(validate_authn_hash) {
         BOOST_REQUIRE_NE(authn1.key(), authn2.key());
     }
 }
+
+BOOST_AUTO_TEST_CASE(make_api_activity_event_authorized) {
+    const ss::socket_address client_addr{ss::ipv4_addr("10.0.0.1", 12345)};
+    const ss::socket_address server_addr{ss::ipv4_addr("10.1.1.1", 23456)};
+    const ss::sstring url = "/v1/test?param=val";
+    const ss::sstring method = "GET";
+    const ss::sstring protocol_name = "https";
+    const ss::sstring host_name = "local";
+    const ss::sstring version = "1.1";
+    const ss::sstring user_agent = "Mozilla";
+    const ss::sstring username = "test";
+
+    auto req = ss::http::request::make(method, host_name, url);
+    req.parse_query_param();
+    req._client_address = client_addr;
+    req._server_address = server_addr;
+    req._version = version;
+    req._headers["User-Agent"] = user_agent;
+    req._headers["Authorization"] = "You shouldn't see this at all";
+    req.protocol_name = protocol_name;
+
+    request_auth_result auth_result{
+      security::credential_user{username},
+      security::credential_password{"password"},
+      "sasl",
+      request_auth_result::superuser::no};
+
+    auto api_activity = sa::make_api_activity_event(
+      req, auth_result, true, std::nullopt);
+
+    auth_result.pass();
+
+    const ss::sstring expected{
```

Comment:
Can we use `fmt::format` with `fmt::arg` to mane this more readable?

Example usage from the docs:

```
fmt::print("Hello, {name}! The answer is {number}. Goodbye, {name}.",
           fmt::arg("name", "World"), fmt::arg("number", 42));
```